pandas/tests/indexes/datetimes/test_datetimelike.py

```diff
@@ -76,3 +76,27 @@ def test_union(self):
         for case in cases:
             result = first.union(case)
             assert tm.equalContents(result, everything)
+
+    def test_add_dti_td(self):
```

Comment:
this is almost the same as special case of the above test in `datetimelike.py` for `DatetimeIndex`, if that test looks good I will remove this one.

Reply:
see my comments above you can simply re-define create_index to make this work (in another test class)